src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    let newState = { ...stateCopy };
```

Comment:
Small improvement: you can avoid of creation of a variable in function every time
In case 'addProperties' you can assign data to stateCopy 
```
Object.assign(stateCopy, action.extraData);
```
In cases 'removeProperties' and 'clear' just replace `newState` with `stateCopy` and then push copy
```
 resultArray.push({ ...stateCopy });
```